### PR TITLE
TR-1455 - Remove old templates implementation, replace with new implementation

### PR DESCRIPTION
**Pull request by nicklemmon**
[TR-1455](https://jira.int.messagesystems.com/browse/TR-1455)

### What Changed
- Removed files in `pages/templates`
- Moved files from `pages/templatesV2` to `pages/templates`
- Removed lead text from new templates list page

### How To Test

- Verify that unused templates files and actions are no longer in our repo
- Verify that `/templates` takes you to the new templates view rather than the old one
- Verify that lead text is now missing from the new templates page
- Verify that the A/B testing feature still works as it relies on some templates actions
- Go through all relevant flows in the new templates editor:
  1. Run end to end tests locally! `npm run test-e2e` and select 'templates'
  1. Make a new template, save it as a draft, then publish it - write HTML, AMP HTML, plain text, and test data and verify the preview renders appropriately
  1. Send a test preview to your email address and verify the content comes through appropriately
  1. Make a change to a template without saving, go to template settings and save, verify that your changes were also saved
  1. Verify that templates can be duplicated and deleted from the list page
  1. Verify that templates can be duplicated and deleted from the edit page
  1. Verify that snippets can be inserted in the template via the kebab icon in the edit page
  1. Verify that AMP boilerplate is inserted via the kebab icon in the edit page on the AMP HTML tab
  1. Verify that the device width updates using the desktop and mobile icon buttons

### To Do

- [ ] Incorporate feedback

### CODE REVIEW

**Comment by PatrickSison** [comment]
I realized there's an obsolete file, `src/config/navItems/campaign.js`

File: src/pages/templates/components/FromEmail.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ class FromEmail extends Component {
   }, 300);
 
   render() {
-    const { domains, value, ...inputProps } = this.props;
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply src/pages/templates/components/FromEmail.js]: So if they aren't being used, is that a problem to remove the reference? I guess I'm not sure what the risk is.

[the remark this answers is not in the document]
Comment (PatrickSison) [reply src/pages/templates/components/FromEmail.js]: I don't think there's a risk. Looking at it, it's just getting passed down to the field, and it's not being used, so it should be ok

**Comment by nicklemmon** [comment]
@PatrickSison Ready for another pass when you have time - I removed that file that isn't in use. Leaving `localForage` for now until we can (eventually) deploy a ticket that will clear everyone's data. After doing that for a certain time period, we'll then remove it.

Thanks! 👍

**Pull request merged**